### PR TITLE
Add basic, stage and notification configuration in configuration section

### PR DESCRIPTION
**Pull request by sanjeevkpandit**
Includes:
- Basic settings
- Stages settings
- Notifications settings

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -17,6 +17,122 @@ Comming soon ;)
...
+#boss.yml
+project_name: PROJECT_NAME
+project_description: PROJECT_DESCRIPTION
+repository_url: https://github.com/<username>/<project-name>
+user: USERNAME
+port: SERVER_PORT
+```
+
+### Stages
+This includes different instances of your application like `dev`, `qa`, `uat`, `staging`, `production`.
```
Comment (kabirbaidhya) [inline README.md]: Stages refer to the configured remote servers where you'd like to deploy your application eg: `dev`, `qa`, `uat`, `staging`, `production`.

File: README.md [hunk 2]
```diff
@@ -17,6 +17,122 @@ Comming soon ;)
...
+        public_url: http://your-app.com
+        app_dir: /path/to/your/app
+        logging:
+            files:
+                - /path/to/error/log/file
+                - /path/to/access/log/file
+```
+
+### Notifications
+This includes the settings for notifying on deployment start and completion. Currently, only hipchat and slack are supported. Further integrations are welcome through pull-requests.
```
Comment (kabirbaidhya) [inline README.md]: `This includes the ...` has been a pattern for each section. :smile: . It'd be better if we can have something different.

Comment (kabirbaidhya) [inline README.md]: Something like this: You can configure to be notified when deployment starts or succeeds. Currently, only hipchat and slack notifications are supported. Further integrations are welcome through pull-requests.

File: README.md [hunk 3]
```diff
@@ -17,6 +17,122 @@ Comming soon ;)
...
+        app_dir: /path/to/your/app
+        logging:
+            files:
+                - /path/to/error/log/file
+                - /path/to/access/log/file
+
+notifications:
+    slack:
+        enabled: true
+        endpoint: SLACK_TOKEN
```
Comment (kabirbaidhya) [inline README.md]: we can use environment variable here like this `${SLACK_ENDPOINT}`

File: README.md [hunk 4]
```diff
@@ -17,6 +17,122 @@ Comming soon ;)
...
+    hipchat:
+        enabled: true
+        notify: true
+        company_name: HIPCHAT_COMPANY_NAME
```
Comment (kabirbaidhya) [inline README.md]: same here

**Comment by kabirbaidhya** [comment]
@sanjeevkpandit When you're done with this can you point this PR to `zero.x` branch instead of `master`? Because `master` has breaking changes and we not release it any sooner. But we can do minor releases on the `zero.x` branch

**Comment by sanjeevkpandit** [comment]
@kabirbaidhya sure.

**Comment by sanjeevkpandit** [comment]
Closing this, created another #37 comparing with `zero.x`

**Pull request closed without merging**